dot/network/notifications.go

```diff
@@ -226,14 +234,29 @@ func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtoc
 			return
 		}
 
-		hs, err := s.readHandshake(stream, decodeBlockAnnounceHandshake)
-		if err != nil {
-			logger.Trace("failed to read handshake", "protocol", info.protocolID, "peer", peer, "error", err)
+		hsTimer := time.NewTimer(handshakeTimeout)
+		defer hsTimer.Stop()
```

Comment:
just wondering why this was chosen vs `time.After`? I guess since you can stop this if the response arrives first?

Reply:
1) With `timer` we can cancel the timer if the response arrives first.
2) I noticed that we need to stop the timer only when the response arrives first, if the `<-timer.C` comes first then we don't need to use stop since we are not resetting the timer.